assets/src/domain/eventEditor/ui/datetimes/datesList/actionsMenu/dropdown/DateMainMenu.tsx

```diff
@@ -7,18 +7,20 @@ import CopyDate from './CopyDate';
 import EditDate from './EditDate';
 import TrashDate from './TrashDate';
 
-const DateMainMenuItem: React.FC = () => {
+import { DateMainMenuProps } from './types';
+
+const DateMainMenu: React.FC<DateMainMenuProps> = ({ datetime }) => {
 	const toggleProps: DropdownToggleProps = {
 		tooltip: __('event date main menu'),
 	};
 
 	return (
 		<DropdownMenu toggleProps={toggleProps}>
-			<EditDate />
-			<CopyDate />
-			<TrashDate />
+			<EditDate datetime={datetime} />
+			<CopyDate datetime={datetime} />
+			<TrashDate datetime={datetime} />
```

Comment:
### 👍 
personally I prefer passing props if the receiving component actually uses it. 
using a context provider is only valid (imho) to prevent having to "courier" a prop through multiple components that don't require it themselves.